### DISCUSSION TITLE
NRPN function example

### DISCUSSION THREAD

**Discussion by Taurar** [FortySevenEffects/arduino_midi_library#256] [Q&A]
This library is incredibly versatile and much appreciated!

I'm an beginner and I'm having trouble figuring out how to use it to send out NRPN values using the NRPN function in the library.  Google has little useful information and I have not seen any examples using e.g. beginNrpn() and  sendNrpnValue() 

Would it be possible to post a short simple example on how to send out NRPN, for example in response to CC's?

Thanks!

**Comment by franky47** [comment]
You can read more about RPN/NRPN here:
http://www.somascape.org/midi/tech/spec.html#rpns

The following example shows how to listen to RPN/NRPN, but because of their complexity, it's not built-in the library, and how to send them:
https://github.com/FortySevenEffects/arduino_midi_library/blob/master/examples/RPN_NRPN/RPN_NRPN.ino

The key part for sending is at the end:

```c++
// Send a RPN sequence (Pitch Bend sensitivity) on channel 1
const midi::Channel channel = 1;
const byte semitones = 12;
const byte cents     = 42;
MIDI.beginRpn(midi::RPN::PitchBendSensitivity, channel);
MIDI.sendRpnValue(semitones, cents, channel);
MIDI.endRpn(channel);
```
